onnx/reference/ops/op_dequantize_linear.py

```diff
@@ -54,26 +60,31 @@ def _run(
             raise RuntimeError("Input 2 must be a vector or a number.")
 
         x_type = self.get_x_type(x)
-        if x_zero_point is not None:
+        f8_type = x_type in {
+            TensorProto.FLOAT8E4M3FN,
+            TensorProto.FLOAT8E4M3FNUZ,
+            TensorProto.FLOAT8E5M2,
+            TensorProto.FLOAT8E5M2FNUZ,
+        }
+        if x_zero_point is not None and not f8_type:
             zero_type = self.get_x_type(x_zero_point)
             if x_type != zero_type:
                 raise RuntimeError(
                     f"Type mismatch {x_type} != {zero_type} in DequantizeLinear."
                 )
-            if x_type in (
-                TensorProto.FLOAT8E4M3FN,
-                TensorProto.FLOAT8E4M3FNUZ,
-                TensorProto.FLOAT8E5M2,
-                TensorProto.FLOAT8E5M2FNUZ,
-            ):
-                raise RuntimeError(
-                    f"x_zero_point not supported for float 8 type {x_type}."
-                )
 
             dx = x.astype(np.float32) - DequantizeLinear.reshape_input(
                 x_zero_point, x.shape, axis
             )
         else:
+            if (
+                f8_type
+                and x_zero_point is not None
+                and x_zero_point.astype(np.uint8) != np.uint8(0)
+            ):
+                raise RuntimeError(
+                    "x_zero_point is not null but should be null for float 8 types."
+                )
```

Comment:
"x_zero_point, if not null, should be 8-bit for float 8 types."

Reply:
8 bit but it should equal to zero. I can say it should a tensor of one element equal to zero.

Reply:
The operator documentation says
```
In the case of dequantizing int32, there’s no zero point (zero point is supposed to be 0). zero-point is
usually not used in the case of float8e4m3fn, float8e4m3fnuz, float8e5m2, float8e5m2fnuz quantization,
but the dequantization formula remains the same for consistency and ‘x_scale’ still determines the output
type.
```
I find this a bit confusing. "usually not used" is not very precise. Can we first agree on what the specification is? And then we can update the documentation to be clear? There are several possibilities:

1. Zero-point is not allowed (as an input) for 8-bit float.
2. Zero-point may be optionally specified, but must be zero if specified.
3. A non-zero zero-point value may be specified (even though it is uncommon).

Reply:
@yuanyao-nv , any comments on this? Thanks!

Reply:
The PR seems to imply it is case 2. In this case, however, it is better to check if all-values in zero-point tensor are zero (instead of expecting it to be a one-element tensor, because the documentation for zero-point says
```
Zero point for input ‘x’. Shape must match x_scale.
```

Reply:
Added multiscale case.